Fail to destroy Cosmos DB account since unable to list write keys

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

* Terraform v0.11.11
* AzureRM v1.21.0

### Description
Sometimes I got error when I delete Cosmos DB account. Not always. The error tend to occur in multi-region configuration it need long operation time. After I get the error and wait several minutes, I can see that the db account has been deleted on Azure Portal. So after that, terraform destroy command complete successfully.

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_cosmosdb_account`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
provider "azurerm" {
  version = "=1.21.0"
}

resource "azurerm_resource_group" "shared" {
  name     = "${var.my_prefix}-my-${var.my_chap}-rg"
  location = "${var.my_resource_group_location}"
}

resource "azurerm_cosmosdb_account" "shared" {
  name                = "${var.my_prefix}-my-${var.my_chap}-db"
  location            = "${azurerm_resource_group.shared.location}"
  resource_group_name = "${azurerm_resource_group.shared.name}"
  offer_type          = "Standard"
  kind                = "MongoDB"

  enable_automatic_failover = true

  consistency_policy {
    consistency_level       = "BoundedStaleness"
    max_interval_in_seconds = 10
    max_staleness_prefix    = 200
  }

  geo_location {
    location          = "${azurerm_resource_group.shared.location}"
    failover_priority = 0
  }

  geo_location {
    location          = "${var.my_failover_location}"
    failover_priority = 1
  }
}
```

### Expected Behavior

Running terraform destroy should successfully delete the terraform provisioned Cosmos DB account on the first attempt.

### Actual Behavior

```
Error: Error refreshing state: 1 error(s) occurred:

* azurerm_cosmosdb_account.shared: 1 error(s) occurred:

* azurerm_cosmosdb_account.shared: azurerm_cosmosdb_account.shared: [ERROR] Unable to List Write keys for CosmosDB Account my-db: documentdb.DatabaseAccountsClient#ListKeys: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="NotFound" Message="Entity with the specified id does not exist in the system.\r\nActivityId: myid, Microsoft.Azure.Documents.Common/2.1.0.0"
```

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform destroy`

--- a/azurerm/resource_arm_cosmos_db_account.go
+++ b/azurerm/resource_arm_cosmos_db_account.go
@@ -601,6 +601,12 @@ func resourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) er
 	// implying that it also returns the read only keys, however this appears to not be the case
 	keys, err := client.ListKeys(ctx, resourceGroup, name)
 	if err != nil {
+		if utils.ResponseWasNotFound(keys.Response) {
+			log.Printf("[DEBUG] Keys were not found for CosmosDB Account %q (Resource Group %q) - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
 		return fmt.Errorf("[ERROR] Unable to List Write keys for CosmosDB Account %s: %s", name, err)
 	}
 	d.Set("primary_master_key", keys.PrimaryMasterKey)
@@ -608,6 +614,12 @@ func resourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) er
 
 	readonlyKeys, err := client.ListReadOnlyKeys(ctx, resourceGroup, name)
 	if err != nil {
+		if utils.ResponseWasNotFound(keys.Response) {
+			log.Printf("[DEBUG] Read Only Keys were not found for CosmosDB Account %q (Resource Group %q) - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
 		return fmt.Errorf("[ERROR] Unable to List read-only keys for CosmosDB Account %s: %s", name, err)
 	}
 	d.Set("primary_readonly_master_key", readonlyKeys.PrimaryReadonlyMasterKey)
@@ -615,8 +627,15 @@ func resourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) er
 
 	connStringResp, err := client.ListConnectionStrings(ctx, resourceGroup, name)
 	if err != nil {
+		if utils.ResponseWasNotFound(keys.Response) {
+			log.Printf("[DEBUG] Connection Strings were not found for CosmosDB Account %q (Resource Group %q) - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
 		return fmt.Errorf("[ERROR] Unable to List connection strings for CosmosDB Account %s: %s", name, err)
 	}
+
 	var connStrings []string
 	if connStringResp.ConnectionStrings != nil {
 		connStrings = make([]string, len(*connStringResp.ConnectionStrings))

